external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala

```diff
@@ -141,10 +143,9 @@ private[spark] class DirectKafkaInputDStream[K, V](
           tp -> Math.max(offset - currentOffsets(tp), 0)
         }
         val totalLag = lagPerPartition.values.sum
-
         lagPerPartition.map { case (tp, lag) =>
           val maxRateLimitPerPartition = ppc.maxRatePerPartition(tp)
-          val backpressureRate = lag / totalLag.toDouble * rate
+          var backpressureRate = lag / totalLag.toDouble * rate
```

Comment:
Why was this changed to a var?

Reply:
Sorry that was a mistake from another change I was trying to do. Forgot to revert. will fix this